snovault/tests/test_invalidation_scope.py

```diff
@@ -324,19 +336,34 @@ def test_invalidation_scope_positive_diffs(self, testapp, test_link_source_schem
         with invalidation_scope_mocks(test_link_source_schema, embedded_list):
             self.run_test_and_reset_secondary(registry, diff, invalidated, secondary, 1)
 
-    @pytest.mark.parametrize('diff,embedded_list,base_types', [
-        ([ITEM_C + '.value'], ['link_one.value'], [ITEM_D])
+    @pytest.mark.parametrize('diff,embedded_list,base_types,child_types', [
+        ([ITEM_D + '.value'], ['link_one.value'], [ITEM_C], [ITEM_D])
     ])
-    def test_invalidation_scope_base_types(self, testapp, diff, embedded_list, base_types, test_parent_type_schema):
+    def test_invalidation_scope_base_types(self, testapp, diff, embedded_list, base_types, child_types,
+                                           test_parent_type_schema):
         """ Runs some tests that involve base type resolution """
         registry = testapp.app.registry
         invalidated = [
             (UUID1, ITEM_A)
         ]
         secondary = {UUID1}
-        with invalidation_scope_mocks(test_parent_type_schema, embedded_list, base_types=base_types):
+        with invalidation_scope_mocks(test_parent_type_schema, embedded_list, base_types=base_types,
+                                      child_types=child_types):
```

Comment:
With complicated args like these, I recommend using keywords for all the args. You could force that in `invalidation_scope_mocks` by doing:
```
def invalidation_scope_mocks(*, schema, embedded_list, base_types=None, child_types=None, use_default_diff=False):
```
and then here do (untested):
```suggestion
        with invalidation_scope_mocks(schema=test_parent_type_schema, embedded_list=embedded_list,
                                      base_types=base_types, child_types=child_types):
```
It's not even that it adds a lot of information, but it feels like the kind of thing that could need further changes and would be susceptible to getting out of alignment, or where you could pass the pretty-arbitrary arguments in the wrong order or whatever.